Source/DafnyCore/UndisposableTextWriter.cs

```diff
@@ -0,0 +1,247 @@
+using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Dafny;
+
+public class UndisposableTextWriter : TextWriter {
```

Comment:
What's the story behind this? I could search through the changes but I think it deserves a comment. :)

Reply:
It's currently unused, only referenced in commented code:
```
      // Code that shouldn't be needed, but prevents some exceptions when running the integration tests in parallel
      // outputWriter = new UndisposableTextWriter(outputWriter);
      // errorWriter = new UndisposableTextWriter(errorWriter);
      // outputWriter = TextWriter.Synchronized(outputWriter);
      // errorWriter = TextWriter.Synchronized(errorWriter);
```

Basically, one exception that when running tests in parallel is usage of already disposed writers, so somehow the tests were sharing writers, and this wrapper prevents them from being disposed which removed those crashes, but it's a patch for problem that needs a better solution.

Reply:
I could remove this since it's currently not used, but it can be useful in debugging when we try get this to pass on CI. I'd rather keep it in for now.

Reply:
Yeah once I spotted the commented out code I understood, we can leave it as is for now.